### PR TITLE
Add essential Pages and update information

### PR DESCRIPTION
**Pull request by Bluejutzu**
- **Added following pages**:
  - Quick Links and Vencloud
  - Plugin Submission
  - Code of Conduct and Contribution guide
  - 404.astro
  - Unofficial plugin installation 
- **Fixed**:
  - Typo in `installing/custom-plugins.md`

## Ideas
- Troubleshooting Guide? (Vencord support exists though)
- [Regex](https://discord.com/channels/1015060230222131221/1170034313518125129/1281404105662791731)?

### CODE REVIEW

**Comment by Bluejutzu** [comment]
sidebar currently 
sidebar

File: src/content/docs/contribute.md [hunk 1]
```diff
@@ -0,0 +1,30 @@
+---
+title: Contributing to Vencord
+description: Different ways to contribute for Vencord
+---
+
+Vencord is a community project that welcomes any kind of contribution from **anyone**.
+Contributions can be made in different aspects of Vencord.
+
+### Plugin Creation
+Vencord provides plugins to modify the Discord Client to your liking. If you want to create your own plugin head to [Plugin Development](http://localhost:4321/installing/)
```
Comment (Faf4a) [inline src/content/docs/contribute.md:10]:
```suggestion
Vencord provides plugins to modify the Discord Client to your liking. If you want to create your own plugin head to [Plugin Development](/installing/)
```

**Review by Faf4a — CHANGES_REQUESTED** [review]

File: src/content/docs/plugins/submission.md [hunk 2]
```diff
@@ -2,3 +2,54 @@
...
+:::caution[Warning]
+Pull requests can be made to the `dev` and `main` branch of [Vencord](https://github.com/Vendicated/Vencord). Unless you are an advanced user, you should prefer the `main` branch. This is because the `dev` branch might contain unstable changes and gets force-pushed frequently, which could cause conflicts in your pull request.
+:::
+
+Now that you have everything ready, you can start making a pull request to Vencord:
+
+If you are new to Git, check [this Guide](https://opensource.com/article/19/7/create-pull-request-github). **That guide explains how to make pull requests!** 
+Make sure that your title mentions you're awaiting approval for a Plugin.
+
+After creating the PR your code will go through checks and needs to get approved before getting pushed into the `main` (or `dev`) build of Vencord.
```
Comment (schoeneBiene) [inline src/content/docs/plugins/submission.md:53]: PRs aren't merged to main

File: src/content/docs/plugins/submission.md [hunk 3]
```diff
@@ -2,3 +2,54 @@
...
+:::
+
+Now that you have everything ready, you can start making a pull request to Vencord:
+
+If you are new to Git, check [this Guide](https://opensource.com/article/19/7/create-pull-request-github). **That guide explains how to make pull requests!** 
+Make sure that your title mentions you're awaiting approval for a Plugin.
+
+After creating the PR your code will go through checks and needs to get approved before getting pushed into the `main` (or `dev`) build of Vencord. 
+
+However, if your PR get's denied don't worry. Vencords Maintainers give comments on approval/denial, so you can always fix any issues. Unless your plugin is in violation of rules.
```
Comment (schoeneBiene) [inline src/content/docs/plugins/submission.md:55]:
```suggestion
However, if your PR gets denied, don't worry. Vencords Maintainers give comments on approval/denial, so you can always fix any issues. Unless your plugin is in violation of rules.
```

File: src/content/docs/advanced/utils/datastore.mdx [hunk 4]
```diff
@@ -0,0 +1,44 @@
+---
+title: DataStore
+description: How to store data in Vencord
+---
+
+The Datastore is a wrapper around IndexedDB. This can store arbitrarily large data and supports a lot of datatypes (Blob, Map, ...). For a full list, see the [mdn link](https://developer.mozilla.org/en-US/docs/Web/API/Web_Workers_API/Structured_clone_algorithm#supported_types). *This is actually just idb-keyval, so if you're familiar with that, you're golden!*
+
+:::caution[Be aware]
+This should always be preferred over the Settings API if possible, as localstorage has very strict size restrictions and blocks the event loop and make sure your keys are unique (tip: prefix them with ur plugin name) and please clean up no longer needed entries.
```
Comment (schoeneBiene) [inline src/content/docs/advanced/utils/datastore.mdx:9]:
```suggestion
This should always be preferred over the Settings API if possible, as localstorage has very strict size restrictions and blocks the event loop. Make sure your keys are unique (tip: prefix them with your plugin name) and please clean up no longer needed entries.
```

File: src/content/docs/advanced/plugins/optionComponent.mdx [hunk 5]
```diff
@@ -0,0 +1,172 @@
...
+        component: () => {}
+    }
+});
+
+// ...
+`
+}/>
+**The file name has to be .tsx to implement the React Component**
+
+Now to the `component`. You can either build a component directly in it or import it from another file. In this guide we create the component in another file and import it later on.
```
Comment (schoeneBiene) [inline src/content/docs/advanced/plugins/optionComponent.mdx:38]:
```suggestion
Now to the `component`. You can either build a component directly in it or import it from another file. In this guide we will create the component in another file and import it later on.
```

File: src/content/docs/advanced/plugins/optionComponent.mdx [hunk 6]
```diff
@@ -0,0 +1,172 @@
...
+            return (<> <SettingsComponent /> </>);
+        }
+    }
+});
+
+// ...
+`
+} />
+
+Now build your code with `pnpm build` (or `pnpm watch`) and then inject it `pnpm inject`.
```
Comment (schoeneBiene) [inline src/content/docs/advanced/plugins/optionComponent.mdx:118]: You don't need to run inject on every build

```suggestion
Now build your code with `pnpm build` (or `pnpm watch`).
```

File: src/content/docs/advanced/utils/datastore.mdx [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+:::
+
+## Using DataStore in your projects
+
+First import it from `@api/index` (alias `src/api/DataStore/index`).
```
Comment (schoeneBiene) [inline src/content/docs/advanced/utils/datastore.mdx:14]:
```suggestion
First import it from `@api/index` (alias for `src/api/DataStore/index`).
```

File: src/content/docs/contribute.md [hunk 8]
```diff
@@ -0,0 +1,30 @@
+---
+title: Contributing to Vencord
+description: Different ways to contribute for Vencord
+---
+
+Vencord is a community project that welcomes any kind of contribution from **anyone**.
+Contributions can be made in different aspects of Vencord.
+
+### Plugin Creation
+Vencord provides plugins to modify the Discord Client to your liking. If you want to create your own plugin head to [Plugin Development](/installing/)
```
Comment (schoeneBiene) [inline src/content/docs/contribute.md:10]:
```suggestion
Vencord provides plugins to modify the Discord Client to your liking. If you want to create your own plugins, head to [Plugin Development](/installing/)
```

File: src/content/docs/plugins/submission.md (hunk 2 above)
Comment (Bluejutzu) [reply src/content/docs/plugins/submission.md:53]: alright 👍

**Comment by Bluejutzu** [comment]
Ready for review

File: src/content/docs/additional/cloud.md [hunk 9]
```diff
@@ -0,0 +1,68 @@
+---
+title: Vencloud
+description: Vencords own API
```
Comment (taskylizard) [inline src/content/docs/additional/cloud.md:3]: Could do a better description perhaps?

File: src/content/docs/additional/cloud.md [hunk 10]
```diff
@@ -0,0 +1,68 @@
...
+
+First of all, you'll have to clone the [source code](https://github.com/Vencord/Vencloud) to a convenient location:
+
+```sh
+git clone https://github.com/Vencord/Vencloud
+```
+
+### Setting up the Config
+
+Copy the example configuration (`.env.example`) to `.env`. Now open it with your text editor of trust and fill in the configuration values.
```
Comment (taskylizard) [inline src/content/docs/additional/cloud.md:26]: "text editor of trust" sounds weird, can be "text editor of choice"

File: src/content/docs/additional/cloud.md [hunk 11]
```diff
@@ -0,0 +1,68 @@
...
+
+Vencloud is Vencord's API for cloud settings sync!
+
+## Self Hosting
+
+:::caution[Warning]
+Your instance has to be HTTPS capable due to [mixed content restrictions](https://developer.mozilla.org/en-US/docs/Web/Security/Mixed_content) in web browsers.
+:::
+
+We provide a Docker build, so you don't need anything installed besides Docker!
```
Comment (taskylizard) [inline src/content/docs/additional/cloud.md:14]: Should have link to installing docker compose: https://docs.docker.com/compose/install/

File: src/content/docs/additional/cloud.md (hunk 9 above)
Comment (Bluejutzu) [reply src/content/docs/additional/cloud.md]: Done

**Pull request closed without merging**